model/src/pyrenew/latent/infections.py

```diff
@@ -104,8 +106,17 @@ def sample(
 
         gen_int_rev = jnp.flip(gen_int)
 
-        n_lead = gen_int_rev.size - 1
-        I0_vec = jnp.hstack([jnp.zeros(n_lead), I0])
+        if I0.size < gen_int_rev.size:
+            raise ValueError(
+                "Initial infections vector must be at least as long as "
+                "the generation interval."
+                f"Initial infections vector length: {I0.size}, "
+                f"generation interval length: {gen_int_rev.size}."
+            )
+        elif I0.size > gen_int_rev.size:
+            I0_vec = I0[: gen_int_rev.size]
```

Comment:
```suggestion
            I0_vec = I0[-gen_int_rev.size:]
```

Reply:
Minor, but I think more readable to always do `I0_vec = I0[-gen_int_rev.size:]` and not do an `elif`.